test/test_example.py

```diff
@@ -65,3 +65,7 @@ def test_flaky_thing_that_fails_then_succeeds(self):
         self._threshold += 1
         if self._threshold < 1:
             raise Exception("Threshold is not high enough.")
+
+
+def test_function():
```

Comment:
Do we need to apply ``@flaky``? Or does this serve our purposes no matter what?

This name isn't descriptive enough of what we're testing. Perhaps ``test_that_flaky_works_with_top_level_functions``.